src/main/java/edu/ucsb/cs56/mapache_search/controllers/GithubController.java

```diff
@@ -0,0 +1,33 @@
+package edu.ucsb.cs56.mapache_search.controllers;
+
+import org.springframework.stereotype.Controller;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+
+import edu.ucsb.cs56.mapache_search.membership.AuthControllerAdvice;
+import edu.ucsb.cs56.mapache_search.membership.GithubOrgMembershipService;
+
+@Controller
+public class GithubController {
+
+    private Logger logger = LoggerFactory.getLogger(SearchController.class);
+
+    @Autowired
+    private AuthControllerAdvice controllerAdvice;
+
+    @GetMapping("/github")
+    public String gitHub(Model model, OAuth2AuthenticationToken token) {
+        logger.info(controllerAdvice.toString());
+        if(controllerAdvice.getIsAdmin(token) || controllerAdvice.getIsMember(token)){
+            return "github";
+        }
+        else{
+            return "/login";
```

Comment:
I wonder if it would be better to have a redirect to the homepage and a flash message
```suggestion java
           return "redirect:/";
```

Reply:
```java 
if (!role.equals("Admin")) {
            redirAttrs.addFlashAttribute("alertDanger", "You do not have permission to access that page");
            return "redirect:/";
        }
```

Reply:
And add this to the parameters for the controller 
```java
RedirectAttributes redirAttrs
```